app/importers/helpers/record_syncer.rb

```diff
@@ -0,0 +1,108 @@
+  # For syncing between a CSV snapshot and an Insights model, particularly
+# for tracking and then removing records in Insights that have been removed
+# in the CSV snapshot.
+#
+# Callers have to match the CSV row to an Insights record, and describe
+# what Insights records are within the window or scope of the sync.
+class RecordSyncer
+  def initialize(options = {})
+    @log = options.fetch(:log)
+
+    @passed_nil_record_count = 0
+    @invalid_rows_count = 0
+    @unchanged_rows_count = 0
+    @updated_rows_count = 0
+    @created_rows_count = 0
+
+    @marked_ids = []
+  end
+
+  # Given a new or persisted record, with attributes updated in memory
+  # based on the CSV, update it in the Insights database.
+  # Also track that the Insights record still exists in the current CSV snapshot.
+  def validate_mark_and_sync!(insights_record)
+    # Passed nil, something failed upstream
+    if insights_record.nil?
+      @passed_nil_record_count = @passed_nil_record_count + 1
```

Comment:
Small style thing, but maybe we want to use the `@passed_nil_record_count += 1` pattern here. That would reduce the surface for variable name mistakes.

Reply:
(And in the blocks below.)

Reply:
cool, thanks!  for some reason I thought you couldn't do that in Ruby (I think that's `++`).